test/e2e/util/util.go

```diff
@@ -1378,9 +1381,93 @@ func ValidateMetrics(t *testing.T, ctx context.Context, hc *hyperv1.HostedCluste
 	})
 }
 
+func createIngressDNSRecord(t *testing.T, ctx context.Context, client crclient.Client, hostedCluster *hyperv1.HostedCluster, clusterOpts *PlatformAgnosticOptions) {
+	t.Helper()
+	g := NewWithT(t)
+	t.Logf("Creating Ingress DNS Record for HostedCluster %s", hostedCluster.Name)
+
+	guestClient := WaitForGuestClient(t, ctx, client, hostedCluster)
+	defaultIngressRouterService := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "router-default",
+			Namespace: "openshift-ingress",
+		},
+	}
+
+	err := wait.PollUntilContextTimeout(ctx, 10*time.Second, 30*time.Minute, true, func(ctx context.Context) (done bool, err error) {
+		err = guestClient.Get(ctx, crclient.ObjectKeyFromObject(defaultIngressRouterService), defaultIngressRouterService)
+		if err != nil {
+			return false, nil
+		}
+		if len(defaultIngressRouterService.Status.LoadBalancer.Ingress) == 0 {
+			return false, nil
+		}
+		return true, nil
+	})
+	g.Expect(err).ToNot(HaveOccurred(), "failed to get router-default service")
+
+	routerDefaultIP := defaultIngressRouterService.Status.LoadBalancer.Ingress[0].IP
+	g.Expect(routerDefaultIP).ToNot(BeEmpty(), "router-default service does not have an IP")
+
+	// Just for CI testing
+	clusterOpts.AWSPlatform.Credentials.AWSCredentialsFile = "/etc/hypershift-pool-aws-credentials/credentials"
+	region := "us-east-1"
+	awsSession, err := clusterOpts.AWSPlatform.Credentials.GetSession("e2e-openstack-dns-record-on-aws", nil, region)
+	g.Expect(err).ToNot(HaveOccurred(), "failed to create AWS session")
+
+	route53Client := route53.New(awsSession, awsutil.NewAWSRoute53Config())
+	g.Expect(route53Client).ToNot(BeNil(), "failed to create Route53 client")
```

Comment:
what destroys this route53 when the cluster goes away later on?

This seems like it's pretty similar to some of the infra setup stages for aws clusters https://github.com/openshift/hypershift/blob/06ab3404fda04032e097bdc15830c67adb39761e/test/e2e/util/fixture.go#L105

Reply:
nothing yet, this is prototyping, I'll write destroy once I get an actual DNS record to be created.

> This seems like it's pretty similar to some of the infra setup stages for aws clusters
Exactly :confused: and I'm even thinking about using that command even if my HostedCluster is on OpenStack platform.

Reply:
> Exactly 😕 and I'm even thinking about re-using that command even if my HostedCluster is on OpenStack platform.

yeah, if nothing else, you could probably figure out a way to refactor that command in a way that reuses the route53 logic alone.

Reply:
that's exactly my plan :+1: